starknet/src/tests/voting_strategies/erc20_votes.cairo

```diff
@@ -0,0 +1,296 @@
+
+        // Make them self delegate
+        testing::set_contract_address(account0);
+        token_contract.delegate(account0);
+        testing::set_contract_address(account1);
+        token_contract.delegate(account1);
+        testing::set_contract_address(account2);
+        token_contract.delegate(account2);
+        testing::set_contract_address(account3);
+        token_contract.delegate(account3);
+        testing::set_contract_address(account4);
+        token_contract.delegate(account4);
+
+        testing::set_contract_address(OWNER());
+
+        // Return them
+        return array![
+            UserAddress::Starknet(account0),
+            UserAddress::Starknet(account1),
+            UserAddress::Starknet(account2),
+            UserAddress::Starknet(account3),
+            UserAddress::Starknet(account4),
+        ];
+    }
+
+    #[test]
+    #[available_gas(1000000000)]
+    fn works() {
+        let (config, space) = setup_space();
+        let vanilla_execution_strategy = get_vanilla_execution_strategy();
+        let accounts = setup_accounts(space.voting_strategies(1));
+
+        let authenticator = IVanillaAuthenticatorDispatcher {
+            contract_address: *config.authenticators.at(0), 
+        };
+
+        let author = UserAddress::Starknet(contract_address_const::<0x5678>());
+        let mut propose_calldata = array::ArrayTrait::<felt252>::new();
+        author.serialize(ref propose_calldata);
+        vanilla_execution_strategy.serialize(ref propose_calldata);
+        ArrayTrait::<felt252>::new().serialize(ref propose_calldata);
+        ArrayTrait::<felt252>::new().serialize(ref propose_calldata);
+
+        // Create a proposal
+        authenticator.authenticate(space.contract_address, PROPOSE_SELECTOR, propose_calldata);
+
+        // Advance to vote start + 1
+        let current = info::get_block_timestamp();
+        testing::set_block_timestamp(current + config.voting_delay + 1);
+
+        let mut vote_calldata = array::ArrayTrait::<felt252>::new();
+        let voter = *accounts.at(0);
+        voter.serialize(ref vote_calldata);
+        let proposal_id = 1_u256;
+        proposal_id.serialize(ref vote_calldata);
+        let choice = Choice::For(());
+        choice.serialize(ref vote_calldata);
+        let mut user_voting_strategies = array![];
+        user_voting_strategies.append(IndexedStrategy { index: 1, params: array![] });
+        user_voting_strategies.serialize(ref vote_calldata);
+        ArrayTrait::<felt252>::new().serialize(ref vote_calldata);
+
+        // Vote on proposal
+        authenticator.authenticate(space.contract_address, VOTE_SELECTOR, vote_calldata);
+
+        testing::set_block_timestamp(current + config.max_voting_duration);
+
+        // Execute proposal
+        space.execute(1_u256, vanilla_execution_strategy.params);
+    }
+
+    #[test]
+    #[available_gas(1000000000)]
+    #[should_panic]
```

Comment:
we should check panic for the specific error expected , easy to miss bugs otherwise

Reply:
you are 100% correct, and one of the test was indeed not panicking for the correct reason :p

47ad5cc884dace269e852ac3f033b8560bbcc0e5